tf2onnx/graph.py

```diff
@@ -1263,11 +1356,31 @@ def replace_input(self, node, old_input, new_input, input_index=None):
             is_replaced = True
         else:
             raise RuntimeError("Unable to replace input %r into %r for node %r." % (old_input, new_input, node.name))
+
+        to_ops = self._input_to_node_name.get(old_input, None)
+        if to_ops is not None:
+            if node.name in to_ops:
+                # A node may take twice the same entry.
+                to_ops.remove(node.name)
```

Comment:
I think this isn't technically correct if a node has two copies of the same input (node.input = [inp1, inp1]) and you replace just one copy (replace_input(node, inp1, inp2, 0) -> node.input = [inp2, inp1].  node is still a consumer of inp1.  This bug should only occur if input_index is not None since otherwise we replace everything.

Reply:
Also why not use the _unregister helper function?

Reply:
You're right. I did not do it because sometimes replace_input is followed by a call to remove_input, sometimes not. So I did not do it. That means ``_input_to_node_name keeps[input_name]`` contains nodes not using the input anymore. This is not an issue as ``_input_to_node_name keeps[input_name]`` to retrieve all nodes using `input_name`. It was previously done with get_nodes(), even if the new set is bigger than necessary, it is still smaller than get_nodes(). I need to review all calls to replace_input and remove_input to be thorough to call _unregister.

Reply:
I'm mostly concerned that some time later someone will forget to update the index and it will introduce a subtle bug that is hard to catch.